data/mods/gen3/moves.ts

```diff
@@ -550,6 +550,11 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 				this.add('cant', pokemon, 'nopp', randomMove.move);
 				return;
 			}
+			if (randomMove.move === 'rest') {
```

Comment:
This isn't the right place either. The failure condition should be in Rest. Rest _itself_ should be the thing failing. Sleep Talk should successfully be able to call Rest.

Reply:
I do notice that Sleep Talk itself doesn't have a proper message sent on failure either in 3 (https://replay.pokemonshowdown.com/gen3customgame-1802139448), so there may be something in both Sleep Talk and Rest to be fixed? Or maybe it has to do with the move call. Either way, hardcoding Rest somewhere isn't the correct fix.